### DISCUSSION TITLE
Environment per folder setup

### DISCUSSION THREAD

**Discussion by AndrMoiseev** [argoproj/argo-cd#15147] [General]
I'm using environment per folder setup: 
* Single master branch for all environments
* Each environment configuration resides in separate folder
* ApplicationSet generates Application for each environment

So, each environment is fully independent, but deploy commits are on single branch. When I deploy my application in one environment (stage) - I see "Out of sync" status on another environment (prod). It looks quite misleading. Are there any knobs to fix this? 

I expect that environments are totally independent and they actually are, but it seems that ArgoCD uses different reasoning here.

Any thoughts?

**Comment by gaeljw** [comment]
Sounds strange. You'd need to share more details in your setup.

If you look at the reasons of the Out of sync, what do you get? What's the diff?
